### PR TITLE
Add English README

### PR DESCRIPTION
**Pull request by chen3feng**

### CODE REVIEW

File: README-en.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  [object pool](flare/doc/object-pool.md) of [NUMA aware](https://en.wikipedia.org/wiki/Non-uniform_memory_access),
+  [zero-copy buffer](flare /doc/buffer.md), etc.
+- High-quality code. Strict adherence to [Google C++ Code Specification](https://google.github.io/styleguide/cppguide.html),
+  with 80% test coverage
+- Complete [documentation](flare/doc) and [examples](flare/example) and [debugging support](flare/doc/debugging.md),
+  easy to get started quickly
+
+## System Requirements
+
+- Linux 3.10 kernel and above, no other OS supported at this time
```
Comment (0x804d8000) [inline README-en.md:44]: and -> or?

File: README-en.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+- x86-64 processors, aarch64 and ppc64le are also supported, but have not been used in production environments
```
Comment (0x804d8000) [inline README-en.md:45]: "environment" (singular)?

File: README-en.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+- GCC 8 and above compiler
```
Comment (0x804d8000) [inline README-en.md:46]: and -> or

File: README-en.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+## Getting Started
+
+Flare comes out of the box with the required [third-party libraries](thirdparty/), so there is usually no need to install additional dependent libraries. Just pull the code and use it under Linux.
+
+The tarball under `thirdparty/` we store via [Git LFS](https://git-lfs.github.com/), so you need to make sure that `git-lfs` is properly installed before pulling the code.
```
Comment (0x804d8000) [inline README-en.md:52]: "we store" -> "were stored"?

File: README-en.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+### Debugging
+
+We believe that the [debugging](flare/doc/debugging.md) experience is also an important part of the development and maintenance process, and we have made some support for this as follows.
+
+- [GDB plugin for listing pending fibers](flare/doc/gdb-plugin.md)
+- [Support for various Sanitizers](flare/doc/sanitizers.md)
+
+### Testing
+
+To improve the experience of writing a single test, we provide a number of tools for [writing single tests](flare/doc/testing.md).
```
Comment (0x804d8000) [inline README-en.md:72]: "writing a single test" -> "writing unit tests"?

File: README-en.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+These include, but are not limited to.
+
+- [RPC Mock](flare/testing/rpc_mock.h)
+- [Redis Mock](flare/testing/redis_mock.h)
+- [COS Mock](flare/testing/cos_mock.h)
+- [HTTP Mock](flare/testing/http_mock.h)
+- [Non-virtual Function Mock](flare/testing/hooking_mock.h)
+- Some tool methods, etc.
+
+Translated with www.DeepL.com/Translator (free version)
```
Comment (0x804d8000) [inline README-en.md:83]: 我就觉得像是机翻。。

File: README-en.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+### Example
+
+We provided some [examples](flare/example) for reference, the following code snippet is a simple relay service (include both client and service).
+
+```cpp
+#include "thirdparty/gflags/gflags.h"
```
Comment (0x804d8000) [inline README-en.md:90]: We've removed all `thirdparty/` prefixes in our code (so as to support other building systems), unfortunately I forgot to remove it from the sample code. 

Mind to remove the prefix from both `README.md` and `README-en.md`?

File: README-en.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+}  // namespace example
+
+int main(int argc, char** argv) {
+  return flare::Start(argc, argv, example::Entry);
+}
+```
+
+With the builtin [M:N user mode thread](flare/doc/fiber.md) support, requesting outside services from flare with synchronous syntax and using the synchorous inteface of our [builtin  clients for many  popular protocols](flare/net) will not incur peforrmance penalty. For more complex concurrency or synchronous requirements, please refer the [document](flare/doc/fiber.md) for details.
+
+Also，the example `*.flare.pb.h` is generated by [our Protocol Buffers plugin](flare/doc/pb-plugin.md). The interfaces generated this way are easier to use compared to the ones generated by Protocol Buffers' `cc_generic_services`
```
Comment (0x804d8000) [inline README-en.md:149]: Also -> Besides.

Comment (0x804d8000) [inline README-en.md:149]: Besides, is the comma here a full-width (i.e., Chinese) one?

File: README-en.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+
+## More Complex Examples
+
+In practice you will often face scenarios where you need to request multiple backends concurrently, and the following examples shows how to do this in Flare.
```
Comment (0x804d8000) [inline README-en.md:153]: "following examples" -> "following example"

Comment (0x804d8000) [inline README-en.md:153]: shows -> shows you

Comment (0x804d8000) [inline README-en.md:153]: "in Flare" -> "with Flare"?

File: README-en.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+    FLARE_LOG_INFO("Got: {}, {}, {}", *http->body(),
+                   flare::FlattenSlow(cos->bytes), rpc->body());
+  }
+
+  // Now fill `response` accordingly.
+  response->set_body("Great success.");
+}
+```
+
+In this example, we.
```
Comment (0x804d8000) [inline README-en.md:191]: "we." -> "we:"

File: README-en.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+- Initiated three asynchronous requests via three different clients ([HTTP](flare/net/http/http_client.h), [Tencent Cloud COS](flare/net/cos/cos_client.h), RPC).
+- Waiting for all requests to complete synchronously via `flare::fiber::BlockingGet`. Here we only block the user mode thread and there are no performance issues.
+- Print logs to output the response of each service.
+
+For demonstration purposes, we are requesting three heterogeneous services here. You can also request homogeneous, or partially homogeneous and partially heterogeneous services in this way if necessary.
+
+## Contribution
+
+For developers who want to learn more about the internal design of Flare, or who want to participate the development of Flare, there is more technical documentation available under [`flare/doc/`](flare/doc/).
```
Comment (0x804d8000) [inline README-en.md:201]: ..., more documentation can be found under ...

File: README-en.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+
+Please refer to [CONTRIBUTING.md](CONTRIBUTING.md) for more details.
+
+## Performance
+
+Due to the characteristics of the business requirements, we prefer to optimize the smoothness of latency and jitter rather than throughput in the design process, but also try to ensure performance under this premise.
+
+For simple comparison purposes, we have provided [preliminary performance data](flare/doc/benchmark.md).
+
+## Acknowledgements
```
Comment (0x804d8000) [inline README-en.md:211]: Credits

File: README-en.md [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+
+- Our underlying implementation draws heavily on the design of [brpc](https://github.com/apache/incubator-brpc).
+- For the RPC part, [grpc](https://grpc.io/) gave us a lot of inspiration.
+- We rely on a lot of [third party libraries](thirdparty/) from the open source community, standing on the shoulders of giants allows us to develop this project faster and better, and therefore actively give back to the open source community.
+
+We would like to thank all these projects together.
```
Comment (0x804d8000) [inline README-en.md:217]: We'd like to express our sincere gratitude and appreciation to them.

File: README-en.md (hunk 6 above)
Comment (chen3feng) [reply README-en.md:83]: 你说的对啊，DeepL 先处理一遍再检查一下。

**Pull request merged**